pkg/drawio/squareTreeNode.go

```diff
@@ -198,30 +198,44 @@ func (tn *SGTreeNode) children() ([]SquareTreeNodeInterface, []IconTreeNodeInter
 }
 func (tn *SGTreeNode) AddIcon(icon IconTreeNodeInterface) {
 	tn.addIconTreeNode(icon)
-	icon.setSG(tn)
+	icon.addSG(tn)
 }
 
 func (tn *SGTreeNode) NotShownInDrawio() bool { return true }
 
-///////////////////////////////////////////////////////////////////////
-
+// /////////////////////////////////////////////////////////////////////
+// PartialSGTreeNode is the actual square of security group on the canvas
+// PartialSGTreeNode represent one or more security groups
+// for layout reasons, a security group can be represented by more than one PartialSGTreeNode
 type PartialSGTreeNode struct {
 	abstractSquareTreeNode
+	sgs []*SGTreeNode
 }
 
-func newPartialSGTreeNode(parent *SGTreeNode) *PartialSGTreeNode {
-	psg := PartialSGTreeNode{newAbstractSquareTreeNode(parent, parent.name)}
-	parent.partialSgs = append(parent.partialSgs, &psg)
+func newPartialSGTreeNode(sgs []*SGTreeNode) *PartialSGTreeNode {
+	psg := PartialSGTreeNode{newAbstractSquareTreeNode(sgs[0].Parent(), ""), sgs}
+	for _, sg := range sgs {
+		sg.partialSgs = append(sg.partialSgs, &psg)
+	}
 	return &psg
 }
 func (tn *PartialSGTreeNode) children() ([]SquareTreeNodeInterface, []IconTreeNodeInterface, []LineTreeNodeInterface) {
 	return nil, tn.elements, tn.connections
 }
-func (tn *PartialSGTreeNode) DrawioParent() TreeNodeInterface {
-	return tn.Parent().Parent()
+func (tn *PartialSGTreeNode) labels() []string {
+	labels := []string{}
+	for _, sg := range tn.sgs {
+		labels = append(labels, sg.name)
+	}
+	return labels
 }
+
 func (tn *PartialSGTreeNode) Kind() string {
-	return tn.parent.Kind()
+	kind := tn.sgs[0].Kind()
```

Comment:
This doesn't seem to work well. In the html I see `Connectivity graph of s:sg1,sg3`

Reply:
in drawio_test I do not call 	SetKind(string), so its empty.
you can see results at 
pkg/ibmvpc/examples/out/analysis_out/out_iks_workers_large_all_vpcs__with_grouping.html

Reply:
Got it. It would be nicer if we don't use abbreviations in `Kind()`.
That is: better have `Connectivity graph of Security Group` rather than `Connectivity graph of SG`.
Maybe open another issue for this one.